sdk/src/asset_handlers/woff_io.rs

```diff
@@ -0,0 +1,2025 @@
+const C2PA_TABLE_TAG: TableTag = TableTag { data: *b"C2PA", };
+/// Tag for the 'head' table in a font.
+const HEAD_TABLE_TAG: TableTag = TableTag { data: *b"head", };
+
+/// 32-bit font-format identifier.
+enum Magic {
+    /// OpenType - 'OTTO'
+    _OpenType = 0x4f54544f,
+    /// TrueType - FIXED 1.0
+    _TrueType = 0x00010000,
+    /// WOFF 1.0 - 'wOFF'
+    Woff = 0x774f4646,
+    /// WOFF 2.0 - 'wOF2'
+    _Woff2 = 0x774f4632,
+}
+
+/// Used to attempt conversion from u32 to a Magic value.
+impl TryFrom<u32> for Magic {
+    type Error = ();
+
+    /// Tries to convert from u32 to a valid font version.
+    fn try_from(v: u32) -> core::result::Result<Self, Self::Error> {
+        match v {
+            //x if x == Magic::TrueType as u32 => Ok(Magic::TrueType),
+            //x if x == Magic::OpenType as u32 => Ok(Magic::OpenType),
+            x if x == Magic::Woff as u32 => Ok(Magic::Woff),
+            //x if x == Magic::Woff2 as u32 => Ok(Magic::Woff2),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Minimal byte-flopper for moving font header structures and known table
+/// contents in and out of storage.
+trait NetworkByteOrderable {
+    fn from_network(&mut self);
+    fn to_network(&mut self);
+}
+
+/// 'C2PA' font table - in storage
+#[derive(Debug)]
+#[repr(C, packed)]
+#[allow(non_snake_case)]
+struct TableC2PARaw {
+    majorVersion: u16,
+    minorVersion: u16,
+    activeManifestUriOffset: u32,
+    activeManifestUriLength: u16,
+    manifestStoreOffset: u32,
+    manifestStoreLength: u32,
+}
+
+impl NetworkByteOrderable for TableC2PARaw {
+    fn from_network(&mut self) {
+        self.majorVersion = u16::from_be(self.majorVersion);
+        self.minorVersion = u16::from_be(self.minorVersion);
+        self.activeManifestUriOffset = u32::from_be(self.activeManifestUriOffset);
+        self.activeManifestUriLength = u16::from_be(self.activeManifestUriLength);
+        self.manifestStoreOffset = u32::from_be(self.manifestStoreOffset);
+        self.manifestStoreLength = u32::from_be(self.manifestStoreLength);
+    }
+
+    fn to_network(&mut self) {
+        self.majorVersion = u16::to_be(self.majorVersion);
+        self.minorVersion = u16::to_be(self.minorVersion);
+        self.activeManifestUriOffset = u32::to_be(self.activeManifestUriOffset);
+        self.activeManifestUriLength = u16::to_be(self.activeManifestUriLength);
+        self.manifestStoreOffset = u32::to_be(self.manifestStoreOffset);
+        self.manifestStoreLength = u32::to_be(self.manifestStoreLength);
+    }
+}
+
+/// 'C2PA' font table - in storage
+#[derive(Clone)]
+#[allow(non_snake_case)]
+pub struct TableC2PA {
+    /// Major version of the C2PA table record
+    pub majorVersion: u16,
+    /// Minor version of the C2PA table record
+    pub minorVersion: u16,
+    /// Optional URI to an active manifest
+    pub activeManifestUri: Option<String>,
+    /// Optional embedded manifest store
+    manifestStore: Option<Box<[u8]>>,
```

Comment:
I know you copied this from code I wrote, but I seeing this now I realize I should have done a `Option<Vec<u8>>`.

Reply:
np, yeah I recall you `Vec`d that other `Box` I made...